src/git-ship

```diff
@@ -26,6 +26,20 @@ function preconditions {
 }
 
 
+function set_previous_branch {
+  # deleting previous branch
+  if [ "$target_branch_name" == "$INITIAL_PREVIOUS_BRANCH_NAME" ]; then
+    set_branch_as_previous_branch "$MAIN_BRANCH_NAME"
+  # deleting current branch
+  elif [ "$target_branch_name" == "$INITIAL_BRANCH_NAME" ]; then
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  # deleting some other branch
+  else
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  fi
```

Comment:
These two last branches result in the same functionality, can't they be combined?